returnn/torch/engine.py

```diff
@@ -298,6 +303,10 @@ def train_epoch(self):
                 else:
                     total_loss.raw_tensor.backward()
 
+            # update the learning rate per step
+            if self._use_dynamic_lr:
+                self._updater.set_current_step_learning_rate(self.global_train_step)
```

Comment:
As discussed, you would just unconditionally call some `self._updater.set_current_train_step(self.global_train_step)` or so here.

Reply:
I have refactored the code and also renamed the function to set_current_train_step now